backend/packages/Upgrade/src/api/controllers/ExperimentClientController.ts

```diff
@@ -846,9 +862,14 @@ export class ExperimentClientController {
    *          '200':
    *            description: Feature flags list
    */
-  @Get('featureflag')
-  public getAllFlags(@Req() request: AppRequest): Promise<FeatureFlag[]> {
-    return this.featureFlagService.find(request.logger);
+  @Post('featureflag')
+  public async getAllFlags(
+    @Req() request: AppRequest,
+    @Body({ validate: true })
+    experiment: ExperimentAssignmentValidator
+  ): Promise<string[]> {
+    const experimentUserDoc = await this.experimentUserService.getUserDoc(experiment.userId, request.logger);
+    return this.featureFlagService.getKeys(experimentUserDoc, experiment.context, request.logger);
```

Comment:
We should return 404 when the user is not found, like with experiment assignment (and the removed 'keys' endpoint).